src/snowflake/snowpark/modin/pandas/__init__.py

```diff
@@ -346,3 +348,8 @@ def __getattr__(name: str) -> Any:
 modin.core.dataframe.algebra.default2pandas.default.DefaultMethod.register = (
     snowflake.snowpark.modin.core.dataframe.algebra.default2pandas.default.DefaultMethod.register
 )
+
+# Call getOrCreate() at import time on behalf of user so that in a stored procedure,
+# the user does not need to do it explicitly.
+if is_in_stored_procedure():
+    snowflake.snowpark.Session.SessionBuilder().getOrCreate()
```

Comment:
Should we add a TODO to stop doing this once Snowpark always creates the session automatically, with a link to the jira ticket for that?

Reply:
I am not aware that snowpark has plan to create the session automatically. cc @sfc-gh-jdu do we have any plan for this?

The problem here is more for stored sproc, in stored sproc, user is suppose to use the session created by stored sproc which uses specific connection configurations, because the one used by users are not usable within stored sproc. 
Originally, we thought stored sproc will always create the session, but seems it is only creating the session when the session parameter passed. The fix/workaround here seems make sure if snowpark pandas is used in stored sproc, make sure we create the session.
However, since session parameter is a required parameter for stored sproc before, it seems it is almost always creating a session object. I think we can always create a session object in snowpark python stored sproc, i don't see a reason why user will not use a session object with snowpark python stored sproc. cc @sfc-gh-sfan

Reply:
@sfc-gh-yzou Please refer to the other conversation threads.